packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -151,15 +151,24 @@ class Plugin {
   }
 
   static List<String> validatePluginYaml(YamlMap yaml) {
-    if (yaml.containsKey('platforms')) {
-      final int numKeys = yaml.keys.toSet().length;
-      if (numKeys != 1) {
-        return <String>[
-          'Invalid plugin specification. There must be only one key: "platforms", found multiple: ${yaml.keys.join(',')}',
-        ];
-      } else {
-        return _validateMultiPlatformYaml(yaml['platforms'] as YamlMap);
-      }
+
+    final bool usesOldPluginFormat = const <String>{
+      'androidPackage',
+      'iosPrefix',
+      'pluginClass',
+    }.any(yaml.containsKey);
+
+    final bool usesNewPluginFormat = yaml.containsKey('platforms');
+
+    if (usesOldPluginFormat && usesNewPluginFormat) {
+      const String errorMessage =
+          'The flutter.plugin.platforms key cannot be used in combination with the old'
```

Comment:
Do we have docs/wiki/website we could link to in this error message?

Reply:
Added the placeholder link for the new schema documentation (https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin).
We're tracking writing this section in https://github.com/flutter/flutter/issues/44287 I plan to do it this week. I'd like to not wait for the link to be ready before landing this in case there's a chance this will make it to next stable.